lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -153,11 +154,16 @@ export class AuthCache extends BrowserStorage {// Singleton
      */
     getAllAccessTokens(clientId: string, homeAccountIdentifier: string): Array<AccessTokenCacheItem> {
         const results = Object.keys(window[this.cacheLocation]).reduce((tokens, key) => {
-            if ( window[this.cacheLocation].hasOwnProperty(key) && key.match(clientId) && key.match(homeAccountIdentifier)) {
+            if ( window[this.cacheLocation].hasOwnProperty(key) && key.match(clientId) && key.match(homeAccountIdentifier) && key.match(Constants.scopes)) {
                 const value = this.getItem(key);
                 if (value) {
-                    const newAccessTokenCacheItem = new AccessTokenCacheItem(JSON.parse(key), JSON.parse(value));
-                    return tokens.concat([ newAccessTokenCacheItem ]);
+                    try {
+                        const parseAtKey = JSON.parse(key);
+                        const newAccessTokenCacheItem = new AccessTokenCacheItem(parseAtKey, JSON.parse(value));
+                        return tokens.concat([ newAccessTokenCacheItem ]);
+                    } catch (e) {
+                        throw ClientAuthError.createCacheParseError();
```

Comment:
We shouldn't swallow the thown error, but pass it in the error we create. Otherwise we lose context of why the error is occurring. Just pass the error object into the createCacheParseError(e) and print it in the error message string.

Reply:
Agree, would be helpful to at least include the key that was invalid.

Reply:
+1